ledger-tool/src/main.rs

```diff
@@ -2766,6 +2912,93 @@ fn main() {
+                                                .collect();
+
+                                            let instructions = message
+                                                .instructions()
+                                                .iter()
+                                                .map(|ix| {
+                                                    let program_id = accounts
+                                                        [ix.program_id_index as usize]
+                                                        .clone();
+
+                                                    let accounts = ix
+                                                        .accounts
+                                                        .iter()
+                                                        .map(|idx| accounts[*idx as usize].clone())
+                                                        .collect();
+
+                                                    let data = hex::encode(&ix.data);
+
+                                                    Instruction {
+                                                        program_id,
+                                                        accounts,
+                                                        data,
+                                                    }
+                                                })
+                                                .collect();
+
+                                            let execution_results =
+                                                batch.execution_results[no].clone();
+
+                                            let is_simple_vote_tx = tx.is_simple_vote_transaction();
+
+                                            Transaction {
+                                                accounts,
+                                                instructions,
+                                                is_simple_vote_tx,
+                                                execution_results,
+                                            }
+                                        })
+                                        .collect();
+
+                                    let batch = TransactionBatch {
+                                        bank_hash: batch.bank_hash.unwrap().to_string(),
+                                        transactions,
+                                    };
+
+                                    if let Some(recorded_slot) =
+                                        recorded_slots.iter_mut().find(|f| f.slot == slot)
+                                    {
+                                        recorded_slot.transactions.push(batch);
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    // writing the json file ends up with a syscall for each number, comma, indentation etc.
```

Comment:
Hmm, interesting finding. Thinking we should make a similar optimization to the implementation of `write_bank_hash_details_file()`?

Reply:
I've added it there as well